apps/hygeia_web/lib/hygeia_web/helpers/viewer_logging.ex

```diff
@@ -0,0 +1,40 @@
+defmodule HygeiaWeb.Helpers.ViewerLogging do
+  @moduledoc false
+
+  alias Hygeia.CaseContext.Person
+  alias Hygeia.UserContext.User
+
+  @spec log_viewer(
+          request_id :: String.t(),
+          auth :: User.t() | Person.t() | :anonymous,
+          ip_address :: :inet.ip_address(),
+          url :: String.t(),
+          action :: atom,
+          resource :: any
+        ) :: :ok
+  def log_viewer(
+        request_id,
+        auth,
+        ip_address,
+        url,
+        action,
+        %resource_name{} = resource
+      ) do
+    message = %{
+      request_id: request_id,
+      auth:
+        case auth do
+          %{uuid: uuid} -> uuid
+          _other -> auth
```

Comment:
Do a specific match for clarity

Also we need to remember what type of auth it was, therefore I would recommend this:

```suggestion
          %User{uuid: uuid} -> {User, uuid}
          %Person{uuid: uuid} -> {Person, uuid}
          :anonymous -> :anonymous
```